src/Component/Data/ComponentData.php

```diff
@@ -10,6 +10,7 @@ final class ComponentData
 	/**
 	 */
 	public function __construct (
+		public readonly string $uid,
```

Comment:
Should we mark this class as internal so that apps won't directly try to use/instantiate it and be safe(r) re BC Breaks?

Reply:
The constructor should probably marked as internal. Do you want to send a PR?